### DISCUSSION TITLE
DataProvider doesn't recognise Changes in DB Rows

### DISCUSSION THREAD

**Discussion by PGLange** [Budibase/budibase#1937] [Help]
Hi All, 
it's me again :) 
... After the redesign of my app Dashboard, I habe !6! DataProviders at one screen, all load different filtered versions of the same db.  When I change values in the rows that one these data provider renders, the row appears as it should in another DataProvider. BUT even after a reload the row is still in the frist DataProvider. See Picture.

![like1](https://user-images.githubusercontent.com/86222990/124603433-3874a400-de6a-11eb-8fbf-866ff26dc1f4.PNG)

I have this in all browsers and all private windows (Chrome / edge / Firefox / Safari),
Is there a way to tell a DataProvider to fetch data again? ... Even an reopening of the App doesn't help. 

Is it maybe an cache problem in docker?

**Comment by aptkingston** [comment]
Hi @PGLange. Data providers automatically fetch data whenever any action is done that might affect the data it is showing. So if you have 2 providers on the same page referencing the same table, they will both refresh whenever a row changes. 

There might be a problem with this with some of the new SQL work though - where is your data stored? I'm guessing an external SQL database rather than internal tables? Do you have different queries for each data provider, or the same query with different parameters?

**Reply by PGLange** [reply]
Hi @aptkingston,
it's stored internal. I had a look in the database, and the values there are the updated ones. So the first DataProvider somehow remembers this row...

It's the same table, only filters vary. Like:
first DP: 
Purchaser_Responsible equals CurrentUser.Email
creator not equals CurrentUser.Email

Second DP: 
Purchaser_Responsible equals CurrentUser.Email
creator equals CurrentUser.Email

**Reply by aptkingston** [reply]
Thanks for the extra info. What you've done looks totally correct. I think the data provider probably is actually fetching data again, but it's just not filtering correctly for some reason. Could you screenshot the settings panels with each data provider selected, and the filter drawer open? Just so I can see the data provider setting and the filters. That will help me see if there's any configuration problems, before I start trying to reproduce this myself.

**Reply by PGLange** [reply]
@aptkingston 
first list: (where a commit button action changes the value of state to "aknowledge")
![error](https://user-images.githubusercontent.com/86222990/124610121-82608880-de70-11eb-9430-305f5356003d.PNG)
![error2](https://user-images.githubusercontent.com/86222990/124610124-82f91f00-de70-11eb-919c-6d13e9b8f383.PNG)

second list: (just showing the requests this user has sent to others)
![partner1](https://user-images.githubusercontent.com/86222990/124610209-986e4900-de70-11eb-9063-7752eca598d4.PNG)
![partner2](https://user-images.githubusercontent.com/86222990/124610210-9906df80-de70-11eb-8945-492c81a2ad35.PNG)

and third list (where the other user has already commited the request)
![closed1](https://user-images.githubusercontent.com/86222990/124610329-b340bd80-de70-11eb-8162-31eaf3291cd0.PNG)
![closed2](https://user-images.githubusercontent.com/86222990/124610332-b3d95400-de70-11eb-82c8-b8b9ae9faf7c.PNG)

THANK YOU for your time and effort - this is supreme help <3
